dashboard/app/views/levels/_admin.html.haml

```diff
@@ -73,6 +73,14 @@
             = link_to build_script_level_path(script_level), build_script_level_path(script_level)
       %br/
       .project_admin
+      %br/
+      - storage_app_id = storage_decrypt_channel_id(request.url.split("/").last).first
```

Comment:
Pulling the project_id from the url doesn't work quite the way I want it to because urls like `/projects/applab/WqvgATYTdOdmTRr0cDag6g/edit` run into problems. How can I get the project id earlier? I have access to it in the Javascript, but by then it's too late.

Reply:
oh I see, you are needing this in ruby/haml on the server. the easiest way I can think of to work around this is to start with both buttons hidden, and then to show one of them (using javascript) once the page has loaded and dashboard.project.getCurrentId() is available. Apologies for sending you down a wrong path on this one, hopefully this isn't too big of a change from how we talked about doing it before.